internal/run/run.go

```diff
@@ -534,6 +542,12 @@ func (w *Workflow) runSource(ctx context.Context, parentStep *WorkflowStep, id s
 			cliEvent.SourceBlobDigest = blobDigest
 		}
 
+		w.lockfile.Sources[id] = workflow.SourceLock{
```

Comment:
Why did we move this above validateDocument?

Reply:
This is just where we have the information about the digests